Replace Self Link with Uri Factory in Documentation

CaitieM20 opened the pull request:
the documentation is inconsistent.  For the most part it uses URI Factories and not self links.  Corrects the Create Permission example to use URIFactory to be consistent with the rest of the doc

azurecla commented:
Hi __@lorenpaulsen__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        This seems like a small (but important) contribution, so no contribution license agreement is required at this point. Real humans will now evaluate your PR.
    </span>

TTYL, AZPRBOT;

tysonn commented:
@micurd - Could you please review this proposed change from @lorenpaulsen for your AzCopy article in the public repository?

micurd commented:
Thanks @lorenpaulsen, I agree that this is a good change. 

@tysonn - I would like to accept this pr.

Commit:
remove self link from documentation

the documentation is inconsistent.  For the most part it uses URI Factories and not self links.  Corrects the Create Permission example to use URIFactory to be consistent with the rest of the doc
